src/main/scala/cromwell/binding/values/WdlArray.scala

```diff
@@ -25,6 +25,7 @@ case class WdlArray(wdlType: WdlArrayType, value: Seq[WdlValue]) extends WdlValu
   def tsvSerialize: Try[String] = {
     wdlType.memberType match {
       case t: WdlPrimitiveType => Success(value.map(_.valueString).mkString("\n"))
+      case WdlObjectType => WdlObject.tsvSerializeArray(value map { _.asInstanceOf[WdlObject] })
```

Comment:
One could check at WdlArray construction that all objects have the same fields, but it only *really* matters if they have the same fields if you try to serialize it.  There's no reason why you can't have heterogeneous objects in an array if you never try to serialize the array into a TSV file.